client/packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/hooks/useStocktakeLineEdit.ts

```diff
@@ -44,59 +39,9 @@ export const useStocktakeLineEdit = (
       })
     );
 
-  const mapStructuredErrors = (
-    result: Awaited<ReturnType<typeof upsertLines>>
-  ) => {
-    const insertResults = result.batchStocktake?.insertStocktakeLines || [];
-    const updateResults = result.batchStocktake?.updateStocktakeLines || [];
-
-    const errorMessagesMap: { [key: string]: string } = {};
-
-    for (const { response, id } of [...insertResults, ...updateResults]) {
-      // First unset error
-      errorsContext.unsetError(id);
-      // No error
-      if (response.__typename === 'StocktakeLineNode') continue;
-
-      const { error } = response;
-      // Common error for all lines
-      if (error.__typename === 'CannotEditStocktake') {
-        errorMessagesMap[error.__typename] = t('error.not-editable');
-        continue;
-      }
-      // Line specific errors
-      switch (error.__typename) {
-        case 'AdjustmentReasonNotProvided':
-          errorMessagesMap[error.__typename] = t('error.provide-reason');
-          break;
-        case 'AdjustmentReasonNotValid':
-          errorMessagesMap[error.__typename] = t('error.provide-valid-reason');
-          break;
-        case 'StockLineReducedBelowZero':
-          errorMessagesMap[error.__typename] = t(
-            'error.stocktake-has-stock-reduced-below-zero'
-          );
-          break;
-
-        default:
-          noOtherVariants(error);
-      }
-
-      errorsContext.setError(id, error);
-    }
-
-    const errorMessages = Object.values(errorMessagesMap);
-    return {
-      errorMessages: errorMessages.length === 0 ? undefined : errorMessages,
-    };
-  };
-
   const save = async () => {
```

Comment:
Self review comment here would be great (where did it go : )

Reply:
hmm, also previously the context was a bit closer, actually it shouldn't even be context, zustand would be much better i think (that's just opinion though)

Reply:
> Self review comment here would be great (where did it go : )

Ah yes sorry, thank you 🙈 

Keen to chat about this briefly when you get a sec...